Added in test Target Name in classname and removed () from name

FabienLydoire opened the pull request:
Although xcpretty and trainer parse different files to produce the Junit output, the content of the Junit file produced should be identical.

The testcase `classname` and `name` are used to identify uniquely a test across different test runs, for example in third party Tests managers like Xray for JIRA.

In the scenario of transition of xcpretty to trainer, it is required that the tests can be identified in a test run before the transition and a run after the transition.

In the Junit produced by xcpretty : 
- the test classname is the test method classname prefixed by the test target in Xcode
- the test name is the name of the test method in the code, without `()`
In the Junit produced by trainer : 
- the test classname is is the test method classname without prefix
- the test name is the name of the test method in the code, including `()`

This Pull Request includes changes to align the Junit test `classname` and `name` with xcpretty format.

segiddins commented:
This will be a breaking change for anyone who is depending upon the output of Trainer being stable

FabienLydoire commented:
@segiddins As you said, the PR as is would be a breaking change, but as I explained transitioning from xcpretty to trainer is also a breaking change !
Maybe we could enable one format over the other by a parameter ?

FabienLydoire commented:
> This will be a breaking change for anyone who is depending upon the output of Trainer being stable

@segiddins : I pushed a new commit to make the change non breaking.

snatchev commented on lib/trainer/options.rb:
```diff
@@ -33,7 +33,13 @@ def self.available_options
-                                     default_value: true)
+                                     default_value: true),
+        FastlaneCore::ConfigItem.new(key: :xcpretty_naming,
+									 short_option: "-x",
```
Looks like we have some inconsistent indentation. Try `rubocop -a` to fix these with the linter.

FabienLydoire replied:
Thanks for the command ! Fixed it.

snatchev commented on lib/trainer/test_parser.rb:
```diff
@@ -108,18 +108,25 @@ def unfold_tests(data)
           test_name: testable_summary["TestName"],
           duration: testable_summary["Tests"].map { |current_test| current_test["Duration"] }.inject(:+),
           tests: unfold_tests(testable_summary["Tests"]).collect do |current_test|
+			if xcpretty_naming
```
Could we extract this logic into a method ?This is already quite difficult to parse.

snatchev commented on lib/trainer/test_parser.rb:
```diff
@@ -108,18 +108,25 @@ def unfold_tests(data)
               identifier: current_test["TestIdentifier"],
-			  test_group: testable_summary["TargetName"] + "." + current_test["TestIdentifier"].split("/")[0..-2].join("."),
-			  name: current_test["TestName"][0..-3],
+			  test_group: test_group,
```
Rubocop should fix these whitespace issues :)

FabienLydoire replied:
Thanks for the command ! Fixed it.

joshdholtz commented on lib/trainer/options.rb:
```diff
@@ -33,7 +33,13 @@ def self.available_options
+                   env_name: "TRAINER_XCPRETTY_NAMING",
+                                     description: "Produces class name and test name identical to xcpretty naming in junit file",
+                   is_string: false,
+                                     default_value: false)
```
This is a bit nit picky but can we align this config item arguments up with the ones above? 🙃

FabienLydoire commented:
@joshdholtz I did update the PR according to the comments. Can you review again please ?
Thanks !

joshdholtz commented:
@FabienLydoire Code looks good but it looks like the tests are failing if you run `rspec. Do you mind taking a stab at fixing those tests? 😇

FabienLydoire commented:
> @FabienLydoire Code looks good but it looks like the tests are failing if you run `rspec`. Do you mind taking a stab at fixing those tests? 😇

Fixed Unit tests and added test for xcpretty naming
